scripts/node.sh

```diff
@@ -1,838 +1,210 @@
-      # if basename matches, print the hash and basename
-      s == basename { printf "%s  %s\n", hash, basename; }
-   '
-}
+get_version
+color "37" "Latest Harmony release is: $harmony_rel."
 
-verify_checksum() {
-   local dir file checksum_file checksum_for_file
-   dir="${1}"
-   file="${2}"
-   checksum_file="${3}"
-   [ -f "${dir}/${checksum_file}" ] || return 0
-   checksum_for_file="${dir}/${checksum_file}::${file}"
-   extract_checksum "${file}" < "${dir}/${checksum_file}" > "${checksum_for_file}"
-   [ -s "${dir}/${checksum_for_file}" ] || return 0
-   if ! (cd "${dir}" && exec md5sum -c --status "${checksum_for_file}")
-   then
-      msg "checksum FAILED for ${file}"
-      return 1
-   fi
-   return 0
-}
-
-verify_signature() {
-   local dir file 
-   dir="${1}"
-   file="${dir}/${2}"
-   sigfile="${dir}/${2}.sig"
+readonly wrapper_dir="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")/staging/${harmony_rel}"
+VALIDATOR="harmony"
+BTCRELAY="hmy-btcrelay"
+ETHRELAY="hmy-ethrelay"
+mkdir -p "${wrapper_dir}"
 
-   result=$(openssl dgst -sha256 -verify "${outdir}/harmony_pubkey.pem" -signature "${sigfile}" "${file}" 2>&1)
-   echo ${result}
-   if  [[ ${result} != "Verified OK" ]]; then
-	   return 1
-   fi
-   return 0
-}
+arch=$(uname -m)
+arch=${arch/x86_64/amd64}
+arch=${arch/aarch64/arm64}
 
-download_binaries() {
-   local outdir status
-   ${do_not_download} && return 0
-   outdir="${1}"
-   mkdir -p "${outdir}"
-   for bin in $(cut -c35- "${outdir}/md5sum.txt"); do
-      status=0
-      curl -sSf http://${BUCKET}.s3.amazonaws.com/${FOLDER}/${bin} -o "${outdir}/${bin}" || status=$?
-      case "${status}" in
-      0) ;;
-      *)
-         msg "cannot download ${bin} (status ${status})"
-         return ${status}
-         ;;
-      esac
-
-      if $verify; then
-         curl -sSf http://${BUCKET}.s3.amazonaws.com/${FOLDER}/${bin}.sig -o "${outdir}/${bin}.sig" || status=$?
-         case "${status}" in
-         0) ;;
-         *)
-            msg "cannot download ${bin}.sig (status ${status})"
-            return ${status}
-            ;;
-         esac
-         verify_signature "${outdir}" "${bin}" || return $?
-      fi
-      verify_checksum "${outdir}" "${bin}" md5sum.txt || return $?
-      msg "downloaded ${bin}"
-   done
-   chmod +x "${outdir}/harmony" "${outdir}/node.sh"
-   (cd "${outdir}" && exec openssl sha256 $(cut -c35- md5sum.txt)) > "${outdir}/harmony-checksums.txt"
-}
-
-_curl_check_exist() {
-   local url=$1
-   local statuscode=$(curl -I --silent --output /dev/null --write-out "%{http_code}" $url)
-   if [ $statuscode -ne 200 ]; then
-      return 1
-   else
-      return 0
-   fi
-}
-
-_curl_download() {
-   local url=$1
-   local outdir=$2
-   local filename=$3
-
-   mkdir -p "${outdir}"
-   if _curl_check_exist $url; then
-      curl --progress-bar -Sf $url -o "${outdir}/$filename" || return $?
-      return 0
-   else
-      msg "failed to find/download $url"
-      return 1
-   fi
-}
+case "$1" in
```

Comment:
Is it reasonable to have node.sh in another repository?

Reply:
better stay within the main repo IMO.